hirundo/dataset_optimization.py

```diff
@@ -764,6 +772,32 @@ def cancel(self) -> None:
             raise ValueError("No run has been started")
         self.cancel_by_id(self.run_id)
 
+    @staticmethod
+    def archive_run_by_id(run_id: str) -> None:
+        """
+        Archive the dataset optimization run for the given `run_id`.
+
+        Args:
+            run_id: The ID of the run to archive
+        """
+        if not run_id:
+            raise ValueError("No run has been started")
```

Comment:
Since this is a static method, I guess a different Error message is appropriate here?